src/screens/CreateGuild/index.tsx

```diff
@@ -0,0 +1,322 @@
+    )
+    .max(32, 'Grito de Guerra deve ter no máximo 32 caracteres'),
+});
+
+const CreateGuild: React.FC = () => {
+  const navigation = useNavigation();
+  const isFocused = useIsFocused();
+  const dispatch = useAppDispatch();
+  const {
+    control,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<FormData>({ resolver: yupResolver(schema) });
+  const [isLoadingRequest, setIsLoadingRequest] = useState<boolean>(false);
+  const [selectedImage, setSelectedImage] = useState<Image>({} as Image);
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const userSession: FirebaseAuthTypes.User = auth().currentUser!;
+
+  useEffect(() => {
+    let isMounted = true;
+
+    isMounted && isFocused && dispatch(setTitle('Create Guild'));
+
+    return () => {
+      isMounted = false;
+    };
+  }, [isFocused]);
+
+  const uploadBanner = async (): Promise<string | void> => {
+    const filename = selectedImage.filename;
+    const uri = selectedImage.uri;
+
+    try {
+      const imageRef = `${uuidv4()}.${getImageType(filename)}`;
+      const reference = storage().ref(imageRef);
+
+      await reference.putFile(uri);
+
+      return imageRef;
+    } catch (err: any) {
+      Alert.alert(
+        '>.<',
+        firebaseExceptions[err.code] || 'Não foi possível alterar o banner.',
+        [
+          {
+            text: 'Ok',
+          },
+        ],
+      );
+    }
+  };
+
+  const onSubmit = async (data: FormData) => {
+    setIsLoadingRequest(true);
+
+    try {
+      let imageRef: string | void;
+
+      if (Object.keys(selectedImage).length !== 0) {
+        imageRef = await uploadBanner();
+      }
+
+      const { id } = await firestore()
+        .collection('guilds')
+        .add({
+          name: data.name,
+          bannerRef: imageRef ? imageRef : '',
+          description: data.description,
+          members: [userSession.uid],
+          owner: userSession.uid,
+          createdAt: firestore.FieldValue.serverTimestamp(),
+          warCry: data.warCry,
+        });
+
+      await firestore()
+        .collection<ProfileDto>('profiles')
+        .doc(userSession.uid)
+        .update({
+          guild: id,
+        });
+
+      Alert.alert('=D', 'Guild criada com sucesso', [
+        {
+          text: 'Ok',
+          onPress: () => navigation.goBack(),
+        },
+      ]);
+    } catch (err) {
+      Alert.alert(
+        '>.<',
+        'Não foi possível concluir a solicitação, tente novamente mais tarde.',
+        [
+          {
+            text: 'Ok',
+          },
+        ],
+      );
+    } finally {
+      setIsLoadingRequest(false);
+    }
+  };
+
+  const handleGallery = async () => {
+    const permissionStatus = await getPermissions('gallery');
+
+    if (permissionStatus === 'granted') {
+      const imageFromStorage = await getPictureFromStorage();
+      const filename = imageFromStorage.assets?.[0].fileName;
+      const uri = imageFromStorage.assets?.[0].uri;
```

Comment:
```suggestion
      const asset = imageFromStorage?.assets?.pop();
      
      if (!asset) throw new Error('algumacoisa');
      
      const filename = asset?.fileName;
      const uri = asset?.uri;
```

Reply:
I can't throw a Error in this function, because it is called by an interface component, and components doesn't handle errors. I'll show an alert and put a "return" to stop function execution instead.

Reply:
Done. I created a function to handle images from gallery e camera.